### ISSUE TITLE
current_temperature is null

### ISSUE THREAD

**Issue by leinich** [pszafer/home-assistant-bosch-custom-component#1]
First of thank you very much for this custom component.
Today I found some time to play around with your custom component and mostly works.

For  water_heater.dhw1 and climate.hc1 current_temperature is reported null.
The Gateway reports: iCom_Low_NSC_v1 (buderus logamatic km200)

Do you have any idea what the waterheater program HCprogram is for?
operation_list: Off,high,HCprogram,ownprogram

For the integration page it would also be great if you could list the other sensors created.

**Comment by pszafer** [comment]
Do you use Buderus app?
Do you have current temperature there? Are you waiting long enough? I mean more than 3 minutes.

About programs. I'm going to open WIKI and with shared forces we gonna explain all programs for everybody.

Integration page - I could, but for some reason HA community AFAIK doesn't want to include sensors in integration page. I don't know why, but I have to follow these rules

**Comment by pszafer** [comment]
Wiki started - https://github.com/pszafer/home-assistant-bosch-custom-component/wiki/Water-heater

**Comment by leinich** [comment]
> Do you use Buderus app?
> Do you have current temperature there? Are you waiting long enough? I mean more than 3 minutes.

Yes, the componten was up and running for several hours.
This is a screenshot from the Buderus MyDevice App:
![image](https://user-images.githubusercontent.com/5610824/57986701-9e2e2800-7a78-11e9-8d46-2f93c9ae86a6.png)
The value I would expect is listed in the sensor sensor.hotwater_temp

The water_heater in ha looks like:
water_heater.dhw1 | ownprogram | current_temperature: null min_temp: 30 max_temp: 80 temperature: 55 target_temp_high: null target_temp_low: null operation_mode: ownprogram operation_list: Off,high,HCprogram,ownprogram CALENDAR: null friendly_name: Buderus Wasserkessel supported_features: 3
-- | -- | --

> About programs. I'm going to open WIKI and with shared forces we gonna explain all programs for everybody.

Thanks. This already answered another finding regarding the  -3276.8 for sensor.return_temp

> Integration page - I could, but for some reason HA community AFAIK doesn't want to include sensors in integration page. I don't know why, but I have to follow these rules
That is very pitty. Some integrations e.g. the MQTT does this and this is very handy.

**Comment by moustic999** [comment]
Hello,
VorlaufTemperatur is not the same as the Current temperature.
current temperature is the actual temperature in the room where your thermostat is installed. I some case thermostat is installed inside your heater, then current temperature is disabled.

the vorlauf temperature is the temperature of the water inside the heater.
this is given in Home assistant as a sensor :  
```
{
   "id": "/system/sensors/temperatures/supply_t1",
   "name": "Actual supply temp"
}
```

**Comment by Railittle** [comment]
Hi moustic,

I use the componement also and it works great.
I have the same question as leinich but doesn't understand what to do, or what you will do with our requests? 
I have also a sensor which is not working as aspected. sensor.hotwater_temp gives Invalid.
Is this the same sensor as current _temperature?
![hotwater](https://user-images.githubusercontent.com/41593683/59631042-2589bc80-9147-11e9-98ca-ce37486003f8.JPG)

Also I would ask or it is possible to implement the maintenance messages also in HA. 
Geen actuele meldingen means "at this moment no actual messages". 

![questions](https://user-images.githubusercontent.com/41593683/59630651-29690f00-9146-11e9-8e3f-f27f0090e42d.JPG)

Regards,
Little

**Comment by pszafer** [comment]
@Railittle current temperature in water heater is `actualTemp`.
Values in app and HA are totally different, 63C vs 47C.

I think only @moustic999 can elaborate about your installation and what can we do to adjust HA component for your installation.

**Comment by moustic999** [comment]
@Railittle regarding the issue with water temperature. there are many paramters which can be retrieved from the gateway.
I will check which one is the best to use and I will check with @pszafer to adapt if any. 
I know that since the last release of firmware buderus/bosch changed the way some parameter are working and it could be one which we did not find before.

**Comment by moustic999** [comment]
@Railittle In order to understand, I would like some explanatation regarding you heater installation.
which heater and which setup /  where is located the thermostat, 1 or more thermostat and how is the water heated ? is it instantly or via an external boiler (160l, 200l ...)

there is 2 different way to heat water : instantly with very small reserve, model such as Buderus GB162T40 ...
or via an external boiler which keep hot water in a big reserve.

dependant of this, different sensor must be checked.

the sensor :  /system/sensors/temperatures/hotWater_t2 (HA : sensor.hotWater) does not seem to be used in your case, as well as in my case. I suspect it is used only in case of instant water, but I don't have this.

for the boiler the current temperature will be 
/dhwCircuits/dhw1/actualTemp 

not sure why current_temperature is ? now. I will check which value is taken into account for this.

**Comment by Railittle** [comment]
@moustic999 

I have a nefit boiler (300L) and the boiler is heated with my Heatpump. In the boiler there is a sensor and this sensor is connected in my heatpump and is called TW1 in the manual (see picture).

![image](https://user-images.githubusercontent.com/41593683/60919795-ec9cbd80-a296-11e9-9a4c-b41146c1c856.png)


My system is like below. I have one external thermostat in my living room.

![image](https://user-images.githubusercontent.com/41593683/60919953-50bf8180-a297-11e9-9f69-2f1b153803a7.png)

**Comment by pszafer** [comment]
@Railittle , @leinich when HA 0.96 will be update, please update this component (released already, hope there won't be any last minutes changes).
I added extra service to HA bosch.debug_scan which creates json with your system setup (without any personal info like UUID). 
Provide us log with your setup, so we can try add temperature which fulfills your needs.

**Comment by leinich** [comment]
Sorry for the late reply. I finally found time to update.

Please find attached the requested JSON.
Thank you very much for your effort in developing this ha component.
[bosch_scan.zip](https://github.com/pszafer/home-assistant-bosch-custom-component/files/3413710/bosch_scan.zip)

**Comment by pszafer** [comment]
@leinich can you paste print screen of HA states of water_heater and climate.
If possible try to contact me on Discord. Maybe we will figure out what's wrong faster that way.

**Comment by pszafer** [comment]
@Railittle if you are willing to give us your rawscan, please create another issue as your setup is different and this issue is becoming hard to follow.

**Comment by Railittle** [comment]
Sorry i’m on holiday in France. I’m back next week. Then I can help you.

Regards,

R

Verstuurd vanaf mijn privé iPhone

Op 22 jul. 2019 om 14:09 heeft Pawel <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


@Railittle<https://github.com/Railittle> if you are willing to give us your rawscan, please create another issue as your setup is different and this issue is becoming hard to follow.

**Comment by moustic999** [comment]
> Sorry for the late reply. I finally found time to update.
> 
> Please find attached the requested JSON.
> Thank you very much for your effort in developing this ha component.
> [bosch_scan.zip](https://github.com/pszafer/home-assistant-bosch-custom-component/files/3413710/bosch_scan.zip)

I checked the scan, and I don't see any "specialties" in it ?
@leinich Can you explain what is wrong for you ?  because there is multiple case in this topic and I'm a bit confused.

**Comment by pszafer** [comment]
This topic is a little mess and it is based on very old version of component.
Closing it. Feel free to open new one if you still have a problem using this component.
